app/controllers/map_controller.rb

```diff
@@ -16,7 +16,10 @@ def index
   end
 
   def show
-    @node = DrupalNode.find_map_by_slug(params[:name]+'/'+params[:date])
+    @node = DrupalNode.find params[:name]
```

Comment:
I further understood from the call (where it was see to meet you face to face!) that the history module of friendly_id will preserve (but redirect) old map URLs, which is a bit better, but still not ideal.

I also learned that you are asking for *some part* of the URL to be completely unique. Would it not be possible to do (pseudocode only, here):

```ruby
if url.match("/map/")
  @node = DrupalNode.find params[:name] + '/' + params[:date]
else
  @node = DrupalNode.find params[:slug]
end
```